treerec/tskit/parallel_sort.cpp

```diff
@@ -0,0 +1,106 @@
+#ifdef __cplusplus
+#include <tskit/tables.h>
+#include <parallel/algorithm>
+#include <iostream>
+//#include "tbb/parallel_sort.h"
+
+extern "C"{
+#endif
+
+void psort_edges(edge_sort_t *sorted_edges, tsk_size_t n);
+void psort_sites(tsk_site_t *sorted_sites, tsk_size_t num_sites);
+void psort_mutations(tsk_mutation_t *sorted_mutations, tsk_size_t num_mutations);
+
+//Bool based comparator functions for sorting. The old ones were giving me problems so recoded it. Hope that's fine. 
```

Comment:
Probably fine; ideally this sort of thing will get pushed back into tskit, though.  Things that are C++ frosting on top of tskit maybe should be put in a new and different place, outside of the tskit/ subdirectory; @petrelharp?

Reply:
yes, we need to figure out the right way to do this